src/js/providers/tracks-mixin.js

```diff
@@ -254,34 +225,321 @@ define(['../utils/underscore',
+                    _cuesByTrackId = {};
+                }
+                _cuesByTrackId[track.id] = { cues: vttCues, loaded: false};
+                return;
+            }
+            // Cues already added
+            if(_cuesByTrackId[track.id] && _cuesByTrackId[track.id].loaded) {
+                return;
+            }
+
+            var cue;
+            _cuesByTrackId[track.id] = { cues: vttCues, loaded: true };
+
+            while((cue = vttCues.pop())) {
+                textTrack.addCue(cue);
+            }
+        } else {
+            track.data = vttCues;
+        }
+    }
+
+    function _createTrack(itemTrack) {
+        var track;
+        if (_renderNatively) {
+            track = document.createElement('track');
+            track.kind    = itemTrack.kind;
+            track.srclang = itemTrack.language || '';
+            track.label   = itemTrack.label;
+            track.mode    = 'disabled';
+        } else {
+            track = itemTrack;
+            track.data = track.data || [];
+        }
+
+        track.id = itemTrack.default || itemTrack.defaulttrack ? 'default' : '';
+        if (!track.id) {
+            track.id = itemTrack.name || itemTrack.file || ('cc' + _textTracks.length);
+        }
+
+        track.label = track.label || track.name || track.language;
+
+        if (!track.label) {
+            track.label = 'Unknown CC'; // TODO: avoid name collision with embedded Unknown CC track
+            _unknownCount++;
+            if (_unknownCount > 1) {
+                track.label += ' [' + _unknownCount + ']';
+            }
+        }
+
+        return track;
+    }
+
+    function _convertToVTTCues(cues) {
+        var VTTCue = window.VTTCue;
+        var vttCues = _.map(cues, function (cue) {
+            return new VTTCue(cue.begin, cue.end, cue.text);
+        });
+        return vttCues;
+    }
+
+    function _parseCuesFromText(srcContent, track) {
+        require.ensure(['../parsers/captions/vttparser'], function (require) {
+            var VTTParser = require('../parsers/captions/vttparser');
+            var parser = new VTTParser(window);
+            parser.oncue = function(cue) {
+                if (_renderNatively) {
+                    track.addCue(cue);
+                } else {
+                    track.data.push(cue);
+                }
+            };
+
+            parser.onparsingerror = function(error) {
+                _errorHandler(error);
+            };
+
+            parser.onflush = function() {
+                // TODO: event saying track is done being parsed
+            };
+
+            parser.parse(srcContent);
+            parser.flush();
+        }, 'vttparser');
+    }
+
+    function _errorHandler(error) {
+        utils.log('CAPTIONS(' + error + ')');
+    }
+
+    function _nativeRenderingSupported(providerName) {
+        return providerName.indexOf('flash') === -1 && utils.isChrome() || utils.isIOS() || utils.isSafari();
```

Comment:
I know operator precedence takes care of this, but would be more readable if you explicitly add parentheses.